### PR TITLE
Bower Init read defaults from package.json

### PR DESCRIPTION
**Pull request by jisaacks**
This is for Issue #1731

There is one part I think I need some guidance with. I am setting the defaults [here](https://github.com/bower/bower/blob/master/lib/commands/init.js#L78) which works fine for every value except for __name__. which appears to be getting a default value elsewhere before reaching that step. I am unsure where though.

### CODE REVIEW

**Comment by jisaacks** [comment]
FYI, it is failing, but only because it is not getting the default name from `package.json`, per what I was saying above.

**Comment by sheerun** [comment]
@jisaacks I tried to debug it, but I failed to do it completely. Maybe @samccone could take a look?

For sure partly responsible is following line:
https://github.com/bower/bower/blob/4d59d266c196839a7221a9a1518d1d5ab657930c/lib/core/Project.js#L564

But even after reading JSON directly with `readJson` tool, problem seems to sill persist.

Hope it helps.

**Comment by samccone** [comment]
sure thing I will take a look!

**Comment by jisaacks** [comment]
So finally I had a few minutes to look into this some.

* It turns out, that you can set default values to [assume](https://github.com/bower/bower/blob/master/lib/util/readJson.js#L25) as json when no json file exists.
* If you do not assume any data, and no json file is found it will [throw](https://github.com/bower/bower/blob/master/lib/util/readJson.js#L39).

My first thought was to send an empty assume object to prevent the exception, and allow setting all default values in [init/setDefaults](https://github.com/bower/bower/blob/master/lib/commands/init.js#L78). However a [name is required](https://github.com/bower/json/blob/master/lib/json.js#L87) and will throw unless one is assumed.

My next thought was to just move the logic that reads the package.json to where the assumptions are set and assume them all. However this would not consider them if a bower.json file exists but was incomplete (assume only gets used if there is no bower.json, it will not fill missing values to an existing one.)

So unfortunately I had to read from the package.json twice.

Also the integrations tests are passing but the coverage is failing. I am unfamiliar with that, I am assuming it is because I am behind master and new tests were added?

**Comment by jisaacks** [comment]
OK, so I merged in master, and now the coverage is passing but travis just went from passing to failing. The failure is the grunt task `exec:coveralls` I do not believe this is related to my code.

**Comment by jisaacks** [comment]
Come to think of it, if yall think its better, I could move all the npm logic to the read json section and assume it along with the name. That would eliminate the redundant reads. Might make more sense as well, the only caveat would be that any package.json values would be ignored if a bower.json is present no matter if it sets those values or not. But that's probably fine. Let me know what yall think.

**Comment by sheerun** [comment]
It probably changes too much, unfortunately. For example we don't want to accidentially read `dependencies` of package.json as bower's dependencies.

**Comment by jisaacks** [comment]
@sheerun it is explicit about what it sets. Right now it only sets name, version, main, authors, license, description, keywords in the init.js. It only sets the name in project.js. In fact if we moved all the logic to project js the changes would be very few because we would be simply reading package.json and cherry picking specific values to assume. That would be almost no changes to existing code/logic.

File: lib/commands/init.js [hunk 1]
```diff
@@ -77,24 +77,35 @@ function saveJson(project, logger, json) {
 
 function setDefaults(config, json) {
     var name;
+    var npm = {};
     var promise = Q.resolve();
 
+    // Grab values from package.json if it exists
+    if (fs.existsSync(path.join(config.cwd, 'package.json'))) {
```
Comment (stevemao) [inline lib/commands/init.js]: Dont use `fs.existsSync` as it will be deprecated. Just `try` reading the file and catch any error.

**Comment by jisaacks** [comment]
Closing for #1797

**Pull request closed without merging**